pkg/tests/tasks/security/authorization/http_test.go

```diff
@@ -16,105 +16,216 @@ package authorization
+		)
+		oc.WaitSMCPReady(t, meshNamespace, smcpName)
+
+		t.LogStep("Install bookinfo with mTLS")
+		app.InstallAndWaitReady(t, app.BookinfoWithMTLS(ns))
+		ProductPage_URL := app.BookinfoProductPageURL(t, meshNamespace)
+
+		t.NewSubTest("deny all http traffic to bookinfo").Run(func(t test.TestHelper) {
+			t.Cleanup(func() {
+				oc.DeleteFromString(t, ns, DenyAllPolicy)
+			})
+			t.LogStep("Apply policy that denies all HTTP requests to bookinfo workloads")
+			oc.ApplyString(t, ns, DenyAllPolicy)
+
+			t.LogStep("Verify that GET request is denied")
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				Curl.Request(t,
+					ProductPage_URL,
+					nil,
+					assert.ResponseContains("RBAC: access denied"),
+				)
+			})
+		})
+
+		t.NewSubTest("only allow HTTP GET request to the productpage workload").Run(func(t test.TestHelper) {
+			t.Cleanup(func() {
+				oc.DeleteFromString(t, ns, ProductpageGETPolicy)
+				oc.DeleteFromString(t, ns, DenyAllPolicy)
+			})
+			t.LogStep("Apply policies that allow access with GET method to the productpage workload and deny requests to other workloads")
+			oc.ApplyString(t, ns, DenyAllPolicy)
+			oc.ApplyString(t, ns, ProductpageGETPolicy)
+
+			t.LogStep("Verify that GET request to the productpage is allowed and fetching other services is denied")
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				Curl.Request(t,
+					ProductPage_URL,
+					nil,
+					assert.ResponseContains("Error fetching product details"),
+				)
+				Curl.Request(t,
+					ProductPage_URL,
+					nil,
+					assert.ResponseContains("Error fetching product reviews"),
+				)
+			})
+		})
+
+		t.NewSubTest("allow HTTP GET requests to all bookinfo workloads").Run(func(t test.TestHelper) {
+			t.Cleanup(func() {
+				oc.DeleteFromString(t, ns, RatingsGETPolicy)
+				oc.DeleteFromString(t, ns, ReviewsGETPolicy)
+				oc.DeleteFromString(t, ns, DetailsGETPolicy)
+				oc.DeleteFromString(t, ns, ProductpageGETPolicy)
+				oc.DeleteFromString(t, ns, DenyAllPolicy)
+			})
+			t.LogStep("Apply policies that allow HTTP GET requests to all bookinfo workloads")
+			oc.ApplyString(t, ns, DenyAllPolicy)
+			oc.ApplyString(t, ns, ProductpageGETPolicy)
+			oc.ApplyString(t, ns, DetailsGETPolicy)
+			oc.ApplyString(t, ns, ReviewsGETPolicy)
+			oc.ApplyString(t, ns, RatingsGETPolicy)
+
+			t.LogStep("Verify that GET requests are allowed to all bookinfo workloads")
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				Curl.Request(t,
+					ProductPage_URL,
+					nil,
+					assert.ResponseDoesNotContain("RBAC: access denied"),
+				)
+				Curl.Request(t,
+					ProductPage_URL,
+					nil,
+					assert.ResponseDoesNotContain("Error fetching product details"),
+				)
+				Curl.Request(t,
+					ProductPage_URL,
+					nil,
+					assert.ResponseDoesNotContain("Error fetching product reviews"),
+				)
+				Curl.Request(t,
+					ProductPage_URL,
+					nil,
+					assert.ResponseDoesNotContain("Ratings service currently unavailable"),
+				)
```

Comment:
This can be a single request with multiple asserts:

```suggestion
				curl.Request(t,
					ProductPage_URL,
					nil,
					assert.ResponseDoesNotContain("RBAC: access denied"),
					assert.ResponseDoesNotContain("Error fetching product details"),
					assert.ResponseDoesNotContain("Error fetching product reviews"),
					assert.ResponseDoesNotContain("Ratings service currently unavailable"),
				)
```

Reply:
ah, wait.. this doesn't work, because the response is closed after the first assert. We need to refactor this so that it reads the response in the Request function and passes it as a string to the assert checks. Can you do that or should I?

Reply:
I'll do it. I'll add a commit to this PR.

Reply:
>     http_test.go:88: Attempt 30/30:
    http_test.go:89: SUCCESS: string 'Error fetching product details' found in response
    http_test.go:89: FATAL: Failed to read response body: http: read on closed response body
    http_test.go:79: Performing cleanup

Reply:
I added the commit to this PR.